melodies_monet/plots/ozone_sonder_plots.py

```diff
@@ -0,0 +1,230 @@
+    elif altitude_method[0] == 'ground level':
+        ax.plot(O3_OBS,ALT_gl,'-*',color=fill_color_list[0],label = 'OBS:'+model_name_list[0])
+        for i in range(len(O3_MODEL_ALL)):
+            ax.plot(O3_MODEL_ALL[i],ALT_gl, '-*',color=fill_color_list[i+1],label = 'MOD:'+model_name_list[i+1])     
+        plt.title('Comparison at '+str(station_name[0])+' on '+str(release_time)+' UTC')
+        plt.legend()
+        plt.ylim(altitude_range[0],altitude_range[1])
+        plt.xlim(ozone_range[0],ozone_range[1])
+        plt.ylabel('ALT-ground level (km)')
+        plt.xlabel('O3 (ppbv)')
+
+
+#Define ozone sonder, vertical single date plot
+def make_vertical_boxplot_os(df,comb_bx,label_bx,model_name_list,altitude_range,altitude_method,altitude_threshold_list,ozone_range,station_name,release_time,fill_color_list,plot_dict,fig_dict,text_dict):
+    ALT_sl = df['altitude']
+    O3_OBS = comb_bx[comb_bx.columns[0]].to_list()
+    len_combx = np.shape(comb_bx)[1]
+    O3_MODEL_ALL = []
+
+    for i in range(1,len_combx):
+        O3_MODEL = comb_bx[comb_bx.columns[i]].to_list()
+        O3_MODEL_ALL.append(O3_MODEL)
+
+    #release height info,get height of each release site to be substract 
+    df_height = pd.DataFrame({
+         'station':['Boulder, Colorado','Huntsville, Alabama','University of Rhode Island','Trinidad Head, California'],
+         'height':[1.743,0.203,0.021,0.046]
+         })
+    if altitude_method[0] == 'ground level':
+        height_value = df_height.loc[df_height['station']==station_name[0]].values[0][1]
+        ALT_gl = ALT_sl - height_value
+
+    #set default figure size
+    if fig_dict is not None:
+        f,ax = plt.subplots(**fig_dict)
+    else:
+        f,ax = plt.subplots(figsize=(8,8))
+
+    #set default text size
+    def_text = dict(fontsize=20)
+    if text_dict is not None:
+        text_kwargs = {**def_text, **text_dict}
+    else:
+        text_kwargs = def_text
+
+    #set color style
+    obs_dict = dict(color='k', linestyle='-',marker='*', linewidth=2.0, markersize=5.)
+    if plot_dict is not None:
+        plot_kwargs = {**obs_dict, **plot_dict}
+        plot_kwargs.pop('column')
+    else:
+        plot_kwargs = obs_dict
+
+    #Define characteristics of boxplot.
+    boxprops = {'edgecolor': 'k', 'facecolor': 'w','linewidth': 1.5}
+    lineprops = {'color': 'k', 'linewidth': 1.5}
+    boxplot_kwargs = {'boxprops': boxprops, 'medianprops': lineprops,
+                  'whiskerprops':  lineprops, 'capprops': lineprops,
+                  'widths':0.3, 'patch_artist': True,'showfliers': False
+#                  'zorder':0,'showfliers': False
+#                  'showmeans': False,'notch'= False,'showfliers': False
+                  }
+    
+    #Make plot
+    #get position list
+    position_list_1 = altitude_threshold_list[:-1]
+    position_list_2 = altitude_threshold_list[1:]
+    position_list_mid = [(position_list_1[i]+position_list_2[i])/2 for i in range(len(position_list_1))]
+
```

Comment:
Having these plots replicated by altitude type seems redudent to me. Can you instead change this to:

if altitude_method[0] =='sea level':
      alt_p = ALT_sl
      alt_p_name = 'ALT-sea level (km)'
elif altitude_method[0] =='ground level':
      alt_p = ALT_gl
      alt_p_name = 'ALT-ground level (km)'

And then just have one plotting routine that uses these variables. You can probably even condense the calculations above into this logic too. This reduces the complexity of the code.

Reply:
this part is resolved